test/python/transpiler/test_commutative_inverse_cancellation.py

```diff
@@ -0,0 +1,684 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test transpiler pass that cancels inverse gates while exploiting the commutation relations."""
+
+import unittest
+import numpy as np
+from qiskit.test import QiskitTestCase
+
+from qiskit import QuantumCircuit
+from qiskit.transpiler import PassManager
+from qiskit.transpiler.passes import CommutativeInverseCancellation
+
+
+class TestCommutativeInverseCancellation(QiskitTestCase):
```

Comment:
I think it'd be good to have some tests for some edge cases like with parameterized gates in the circuit, some directives like barrier. Also some of the operations explicitly skipped like reset. Just to make sure all those paths are exercised.

Reply:
I have added these corner case tests, and have also added dedicated tests for `CommutationChecker`.